specs/networking/p2p-interface.md

```diff
@@ -0,0 +1,743 @@
+# Overview
+
+This document contains the network specification for Ethereum 2.0 clients.
+
+It consists of four main sections:
+
+1. A specification of the network fundamentals detailing the two network configurations: interoperability test network, and mainnet launch.
+2. A specification of the three network interaction _domains_ of ETH2.0: (a) the gossip domain, (b) the discovery domain, \(c\) the Req/Resp domain.
+3. The rationale and further explanation for the design choices made in the previous two sections.
+4. An analysis of the maturity/state of the libp2p features required by this spec across the languages in which ETH 2.0 clients are being developed.
+
+## Table of Contents
+
+<!-- cmd: doctoc --maxlevel=2 p2p-interface.md -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Network Fundamentals](#network-fundamentals)
+  - [Transport](#transport)
+  - [Encryption and identification](#encryption-and-identification)
+  - [Protocol Negotiation](#protocol-negotiation)
+  - [Multiplexing](#multiplexing)
+- [ETH2 network interaction domains](#eth2-network-interaction-domains)
+  - [Constants](#constants)
+  - [The gossip domain: gossipsub](#the-gossip-domain-gossipsub)
+  - [The discovery domain: discv5](#the-discovery-domain-discv5)
+  - [The Req/Resp domain](#the-reqresp-domain)
+- [Design Decision Rationale](#design-decision-rationale)
+  - [Transport](#transport-1)
+  - [Multiplexing](#multiplexing-1)
+  - [Protocol Negotiation](#protocol-negotiation-1)
+  - [Encryption](#encryption)
+  - [Gossipsub](#gossipsub)
+  - [Req/Resp](#reqresp)
+  - [Discovery](#discovery)
+  - [Compression/Encoding](#compressionencoding)
+- [libp2p Implementations Matrix](#libp2p-implementations-matrix)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Network Fundamentals
+
+This section outlines the specification for the networking stack in Ethereum 2.0 clients.
+
+Sections that have differing parameters for mainnet launch and interoperability testing are split into subsections. Sections that are not split have the same parameters for interoperability testing as mainnet launch.
+
+## Transport
+
+Even though libp2p is a multi-transport stack (designed to listen on multiple simultaneous transports and endpoints transparently), we hereby define a profile for basic interoperability.
+
+#### Interop
+
+All implementations MUST support the TCP libp2p transport, and it MUST be enabled for both dialing and listening (i.e. outbound and inbound connections).
+
+The libp2p TCP transport supports listening on IPv4 and IPv6 addresses (and on multiple simultaneously). Clients SHOULD allow the operator to configure the listen IP addresses and ports, including the addressing schemes (IPv4, IPv6).
```

Comment:
Why `SHOULD`? the ability to configure listen IP/ports sounds like a `MUST` to me

Reply:
actually, I think we can simply remove that sentence. what configuration clients offer is domain-specific to that client.